driver/src/main/java/org/neo4j/driver/internal/net/SocketConnection.java

```diff
@@ -66,10 +66,19 @@ public SocketConnection( BoltServerAddress address, SecurityPlan securityPlan, L
         this.socket.start();
     }
 
-    // for mocked socket testing
-    SocketConnection( SocketClient socket, Logger logger )
+    /**
+     * Create new connection backed by the given socket.
+     * <p>
+     * <b>Note:</b> this constructor should be used <b>only</b> for testing.
+     *
+     * @param socket the socket to use for network interactions.
+     * @param serverInfo the info about server this connection points to.
+     * @param logger the logger.
+     */
+    public SocketConnection( SocketClient socket, InternalServerInfo serverInfo, Logger logger )
     {
         this.socket = socket;
+        this.serverInfo = serverInfo;
```

Comment:
This parameter is needed for `PooledConnectionValidatorTest#sendsSingleResetMessageForValidConnection` test. Field `serverInfo` is usually initialized during `init` and this makes it easier to test just the validation part.